src/query/settings/src/lib.rs

```diff
@@ -125,7 +125,11 @@ impl Settings {
             // Set max memory usage.
             {
                 if ret.get_max_memory_usage()? == 0 {
-                    let max_usage = sysinfo::System::new_all().available_memory();
+                    let max_usage = if conf.query.max_memory_usage == 0 {
+                        sysinfo::System::new_all().available_memory()
```

Comment:
did it work in virtual machine ?

Reply:
No, it will get the host machine's Memory, so we add `max_memory_usage` to the config to throttle.

Reply:
> did it work in virtual machine ?

Virtual machines are OK,  but it would not work with container resources limited by cgroup.